File: SDKs/art19.com/1.0.0/typescript/src/sdk/models/shared/season.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ResourceIdentifier } from "./resourceidentifier";


export enum SeasonAttributesStatusEnum {
    Active = "active",
    Inactive = "inactive"
}


export class SeasonAttributes extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cascaded_cover_image_id" })
  cascadedCoverImageId?: string;

  @SpeakeasyMetadata({ data: "json, name=cover_image_id" })
  coverImageId?: string;

  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=description_is_html" })
  descriptionIsHtml?: boolean;

  @SpeakeasyMetadata({ data: "json, name=description_plain" })
  descriptionPlain?: string;

  @SpeakeasyMetadata({ data: "json, name=first_released_episode_id" })
  firstReleasedEpisodeId?: string;

  @SpeakeasyMetadata({ data: "json, name=last_released_episode_id" })
  lastReleasedEpisodeId?: string;

  @SpeakeasyMetadata({ data: "json, name=season_number" })
  seasonNumber?: number;

  @SpeakeasyMetadata({ data: "json, name=sort_title" })
  sortTitle?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: SeasonAttributesStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;

  @SpeakeasyMetadata({ data: "json, name=updated_at" })
  updatedAt?: Date;
}


// SeasonRelationshipsCascadedCoverImage
/** 
 * The cover art for this season. If there is no specific cover for the season,
 * the cover image of the series' cover art will be returned.
 * The `/images` endpoint provideds details to the image.
 * 
**/
export class SeasonRelationshipsCascadedCoverImage extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: ResourceIdentifier;
}


export class SeasonRelationshipsAListOfClassificationsAssociatedWithTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


// SeasonRelationshipsCoverImage
/** 
 * The cover image for the season
**/
export class SeasonRelationshipsCoverImage extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: ResourceIdentifier;
}


export class SeasonRelationshipsAListOfCreditsGivenToPeopleInTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


export class SeasonRelationshipsAListOfPublishedAndReleasedEpisodesWithAnActualMediaFileWithinTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


// SeasonRelationshipsFirstReleasedEpisode
/** 
 * The first released episode within the season
**/
export class SeasonRelationshipsFirstReleasedEpisode extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: ResourceIdentifier;
}


export class SeasonRelationshipsBothTheCoverImageAndTheCascadedCoverImageForTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


// SeasonRelationshipsLastReleasedEpisode
/** 
 * The most recently released episode within the season
**/
export class SeasonRelationshipsLastReleasedEpisode extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: ResourceIdentifier;
}


export class SeasonRelationshipsTheSeriesTheSeasonBelongsTo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: ResourceIdentifier;
}


export class SeasonRelationshipsAListOfAssociationsLinkingTagsToTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


export class SeasonRelationshipsAListOfTagsAssociatedWithTheSeason extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ResourceIdentifier })
  data?: ResourceIdentifier[];
}


export class SeasonRelationships extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cascaded_cover_image" })
  cascadedCoverImage?: SeasonRelationshipsCascadedCoverImage;

  @SpeakeasyMetadata({ data: "json, name=classification_inclusions" })
  classificationInclusions?: SeasonRelationshipsAListOfClassificationsAssociatedWithTheSeason;

  @SpeakeasyMetadata({ data: "json, name=cover_image" })
  coverImage?: SeasonRelationshipsCoverImage;

  @SpeakeasyMetadata({ data: "json, name=credits" })
  credits?: SeasonRelationshipsAListOfCreditsGivenToPeopleInTheSeason;

  @SpeakeasyMetadata({ data: "json, name=episodes" })
  episodes?: SeasonRelationshipsAListOfPublishedAndReleasedEpisodesWithAnActualMediaFileWithinTheSeason;

  @SpeakeasyMetadata({ data: "json, name=first_released_episode" })
  firstReleasedEpisode?: SeasonRelationshipsFirstReleasedEpisode;

  @SpeakeasyMetadata({ data: "json, name=images" })
  images?: SeasonRelationshipsBothTheCoverImageAndTheCascadedCoverImageForTheSeason;

  @SpeakeasyMetadata({ data: "json, name=last_released_episode" })
  lastReleasedEpisode?: SeasonRelationshipsLastReleasedEpisode;

  @SpeakeasyMetadata({ data: "json, name=series" })
  series?: SeasonRelationshipsTheSeriesTheSeasonBelongsTo;

  @SpeakeasyMetadata({ data: "json, name=taggings" })
  taggings?: SeasonRelationshipsAListOfAssociationsLinkingTagsToTheSeason;

  @SpeakeasyMetadata({ data: "json, name=tags" })
  tags?: SeasonRelationshipsAListOfTagsAssociatedWithTheSeason;
}


export class Season extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=attributes" })
  attributes?: SeasonAttributes;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=relationships" })
  relationships?: SeasonRelationships;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}

File: SDKs/art19.com/1.0.0/python/src/sdk/models/shared/season.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import resourceidentifier as shared_resourceidentifier

class SeasonAttributesStatusEnum(str, Enum):
    ACTIVE = "active"
    INACTIVE = "inactive"


@dataclass_json
@dataclasses.dataclass
class SeasonAttributes:
    cascaded_cover_image_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cascaded_cover_image_id') }})
    cover_image_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cover_image_id') }})
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    description_is_html: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description_is_html') }})
    description_plain: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description_plain') }})
    first_released_episode_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('first_released_episode_id') }})
    last_released_episode_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('last_released_episode_id') }})
    season_number: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('season_number') }})
    sort_title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sort_title') }})
    status: Optional[SeasonAttributesStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    updated_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updated_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsCascadedCoverImage:
    r"""SeasonRelationshipsCascadedCoverImage
    The cover art for this season. If there is no specific cover for the season,
    the cover image of the series' cover art will be returned.
    The `/images` endpoint provideds details to the image.
    
    """
    
    data: Optional[shared_resourceidentifier.ResourceIdentifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsAListOfClassificationsAssociatedWithTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsCoverImage:
    r"""SeasonRelationshipsCoverImage
    The cover image for the season
    """
    
    data: Optional[shared_resourceidentifier.ResourceIdentifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsAListOfCreditsGivenToPeopleInTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsAListOfPublishedAndReleasedEpisodesWithAnActualMediaFileWithinTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsFirstReleasedEpisode:
    r"""SeasonRelationshipsFirstReleasedEpisode
    The first released episode within the season
    """
    
    data: Optional[shared_resourceidentifier.ResourceIdentifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsBothTheCoverImageAndTheCascadedCoverImageForTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsLastReleasedEpisode:
    r"""SeasonRelationshipsLastReleasedEpisode
    The most recently released episode within the season
    """
    
    data: Optional[shared_resourceidentifier.ResourceIdentifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsTheSeriesTheSeasonBelongsTo:
    data: Optional[shared_resourceidentifier.ResourceIdentifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsAListOfAssociationsLinkingTagsToTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationshipsAListOfTagsAssociatedWithTheSeason:
    data: Optional[list[shared_resourceidentifier.ResourceIdentifier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

@dataclass_json
@dataclasses.dataclass
class SeasonRelationships:
    cascaded_cover_image: Optional[SeasonRelationshipsCascadedCoverImage] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cascaded_cover_image') }})
    classification_inclusions: Optional[SeasonRelationshipsAListOfClassificationsAssociatedWithTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('classification_inclusions') }})
    cover_image: Optional[SeasonRelationshipsCoverImage] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cover_image') }})
    credits: Optional[SeasonRelationshipsAListOfCreditsGivenToPeopleInTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('credits') }})
    episodes: Optional[SeasonRelationshipsAListOfPublishedAndReleasedEpisodesWithAnActualMediaFileWithinTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('episodes') }})
    first_released_episode: Optional[SeasonRelationshipsFirstReleasedEpisode] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('first_released_episode') }})
    images: Optional[SeasonRelationshipsBothTheCoverImageAndTheCascadedCoverImageForTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('images') }})
    last_released_episode: Optional[SeasonRelationshipsLastReleasedEpisode] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('last_released_episode') }})
    series: Optional[SeasonRelationshipsTheSeriesTheSeasonBelongsTo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('series') }})
    taggings: Optional[SeasonRelationshipsAListOfAssociationsLinkingTagsToTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('taggings') }})
    tags: Optional[SeasonRelationshipsAListOfTagsAssociatedWithTheSeason] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tags') }})
    

@dataclass_json
@dataclasses.dataclass
class Season:
    attributes: Optional[SeasonAttributes] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attributes') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    relationships: Optional[SeasonRelationships] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('relationships') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
